mayastor-test/csi-e2e/dynamic_provisioning_test.go

```diff
@@ -0,0 +1,264 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	clientset "k8s.io/client-go/kubernetes"
+	restclientset "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"mayastor-csi-e2e/driver"
+	"mayastor-csi-e2e/testsuites"
+)
+
+// TODO: Make configurable
+// was 10Gi
+var smallClaimSize = "50Mi"
+
+// was 100Gi
+var largeClaimSize = "500Mi"
+
+var _ = ginkgo.Describe("Dynamic Provisioning", func() {
+	f := framework.NewDefaultFramework("mayastor")
+
+	tmp := os.Getenv("SMALL_CLAIM_SIZE")
+	if tmp != "" {
+		smallClaimSize = tmp
+	}
+	tmp = os.Getenv("LARGE_CLAIM_SIZE")
+	if tmp != "" {
+		largeClaimSize = tmp
+	}
+
+	var (
+		cs         clientset.Interface
+		ns         *v1.Namespace
+		testDriver driver.PVTestDriver
+	)
+
+	ginkgo.BeforeEach(func() {
+		checkPodsRestart := testCmd{
+			command:  "sh",
+			args:     []string{"mayastor-test/csi-e2e/check_driver_pods_restart.sh"},
+			startLog: "Check driver pods for restarts",
+			endLog:   "Check successful",
+		}
+		execTestCmd([]testCmd{checkPodsRestart})
+
+		cs = f.ClientSet
+		ns = f.Namespace
+
+		var err error
+		_, err = restClient(testsuites.SnapshotAPIGroup, testsuites.APIVersionv1beta1)
+		if err != nil {
+			ginkgo.Fail(fmt.Sprintf("could not get rest clientset: %v", err))
+		}
+	})
+
+	testDriver = driver.InitMayastorDriver()
+	ginkgo.It("should create a volume on demand with mount options [mayastor-csi.openebs.io]", func() {
+		pods := []testsuites.PodDetails{
+			{
+				Cmd: "echo 'hello world' > /mnt/test-1/data && grep 'hello world' /mnt/test-1/data",
+				Volumes: []testsuites.VolumeDetails{
+					{
+						ClaimSize: smallClaimSize,
+						VolumeMount: testsuites.VolumeMountDetails{
+							NameGenerate:      "test-volume-",
+							MountPathGenerate: "/mnt/test-",
+						},
+					},
+				},
+			},
+		}
+		test := testsuites.DynamicallyProvisionedCmdVolumeTest{
+			CSIDriver:              testDriver,
+			Pods:                   pods,
+			StorageClassParameters: defaultStorageClassParameters,
+		}
+
+		test.Run(cs, ns)
+	})
+
+	ginkgo.It("should create multiple PV objects, bind to PVCs and attach all to different pods on the same node [mayastor-csi.openebs.io]", func() {
+		pods := []testsuites.PodDetails{
+			{
+				Cmd: "while true; do echo $(date -u) >> /mnt/test-1/data; sleep 100; done",
```

Comment:
How was this sleep duration determined?

Reply:
Don't know, I've left it unchanged from the original source.